redhat-developer/gitops-operator
# Create console link from the operator

ConsoleLink is a cluster-scoped CRD used by admins to customize OpenShift console. Installing the GitOps operator should create a ConsoleLink resource, which will be useful to navigate to external ArgoCD UI. This PR adds a new controller that can reconcile on ArgoCD instance in `argocd ` namespace and handle ConsoleLink resource.

**Watchers:**

```
1. ArgoCD instance:  Reconcile only for ArgoCD instance named argocd in argocd namespace. We can ignore other instances of ArgoCD
2. ArgoCD server route: Reconcile when the ArgoCD server route is created/updated/deleted and regenerate the ConsoleLink 
```

**Controller Logic:**
```

1. Fetch ArgoCD instance
2. If the instance is absent
        * delete the ConsoleLink if present
3. Fetch the argocd-server route
4. If the route is absent
        * delete the ConsoleLink if present
5. Create the ConsoleLink if absent 

```

**Steps to test:**

1. Install the ArgoCD operator in `argocd` namespace and create an instance named `argocd` with route enabled
2. Install the GitOps operator 
3. Check the ConsoleLink in the `Applications` menu

![Screenshot from 2020-10-05 17-14-59](https://user-images.githubusercontent.com/21128732/95076041-e104a480-072e-11eb-9181-c05ed3f5501b.jpg)
